### PR TITLE
High level objects not always parsed correctly

### PR DESCRIPTION
**Pull request by chrisdp**
This was related to using `exec` with a `g` flag. See the following from the docs:

If your regular expression uses the "g" flag, you can use the exec() method multiple times to find successive matches in the same string. When you do so, the search starts at the substring of str specified by the regular expression's lastIndex property (test() will also advance the lastIndex property).  Note that the lastIndex property will not be reset when searching a different string it will start its search at its existing lastIndex .

### CODE REVIEW

**Comment by chrisdp** [comment]
Has fix for issue #137

File: src/RokuAdapter.ts [hunk 1]
```diff
@@ -639,9 +639,29 @@ export class RokuAdapter {
...
+        return new RegExp(/<.*:\s*(\w+\s*\:*\s*[\w\.]*)>/gi).exec(value);
+    }
+
+    /**
+     * Runs a regex to get the first work of a line
+     * @param value
+     */
+    private getFirstWord(value: string) {
+        return new RegExp(/^([\w.\-=]*)\s/).exec(value);
+    }
```
Comment (chrisdp) [inline src/RokuAdapter.ts]: Changed the above into functions to make it a bit harder to accidentally break a regex that is used in many places by simply adding a g flag.

File: src/RokuAdapter.ts [hunk 2]
```diff
@@ -792,7 +812,7 @@ export class RokuAdapter {
                     return children;
                 }
                 let match;
-                match = /(\w+):(.+)/i.exec(line);
+                match = /(\S+[^:]):(.+)/i.exec(line);
```
Comment (chrisdp) [inline src/RokuAdapter.ts]: This adds support for keys like this:
```
"Food & Both":{...},
"Food&And":{...},
"Food Space":{...}
```

Comment (TwitchBronBron) [reply src/RokuAdapter.ts]: What about keys with colons in them? 

`"my:key"`

Comment (chrisdp) [reply src/RokuAdapter.ts]: While possible I'm not sure how we could detect that because of the way the Roku prints key to the console.

**Pull request merged**